app/api/case/controller.js

```diff
@@ -0,0 +1,191 @@
+ *
+ */
+exports.createCasePoint = async (req, res) => {
+  const { caseId, point } = req.body;
+
+  if (!caseId) throw new Error('Case ID is not valid.')
+  if (!point.latitude) throw new Error('Latitude is not valid.')
+  if (!point.longitude) throw new Error('Latitude is not valid.')
+  if (!point.time) throw new Error('Latitude is not valid.')
+
+  let concernPoint = await casesService.createCasePoint(caseId, point)
+  if (concernPoint) {
+    res.status(200).json({ concernPoint });
+  }
+  throw new Error('Internal server error.');
+};
+
+/**
+ * @method consentToPublish
+ *
+ * Captures user consent to having their data published in the 
+ * aggregated anonymized JSON file that is available to public.
+ *
+ */
+exports.consentToPublish = async (req, res) => {
+  const { caseId } = req.body;
+
+  if (!caseId) throw new Error('Case ID is not valid.')
+
+  const data = {
+    message: 'All Ok!',
+  };
+
+  res.status(200).json(data);
+};
+
+/**
+ * @method setCaseToStaging
+ *
+ * Updates the state of the case from unpublished to staging.
+ *
+ */
+exports.setCaseToStaging = async (req, res) => {
+  const { caseId } = req.body;
+
+  if (!caseId) throw new Error('Case ID is not valid.')
+
+  let caseResults = await casesService.moveToStaging(caseId)
+  if (caseResults) {
+    res.status(200).json({ case: caseResults });
+  }
+  throw new Error('Internal server error.');
+};
+
+/**
+ * @method publishCases
+ *
+ * Moves the state of the cases from staging to published and 
+ * generates JSON file containing aggregated anonymized points 
+ * of concern data. JSON file is then pushed to the endpoint 
+ * responsible for hosting the published data (this functionality 
+ * is implemented by HA).
+ * 
+ * DONE - Fetch Orgnization
+ * DONE - Publish case ids passed in.
+ * DONE - Fetch Points for Cases that are published and have not exipired along with all points.
+ * DONE  -Figure out start and end dates for trails.
+ * DONE - Create Publication
+ * DONE - Build Files
+ * 
+ * Many options when talking about response, and it's all triggered by passing in the type query param.
+ * By default, we will write to a GCS bucket.  Other options include, that do not work in production:
+ * 
+ * zip = Return a zip file
+ * json = Return a JSON payload of what goes into the files that are generated
+ * local = Save to local server environment
+ *
+ */
+exports.publishCases = async (req, res) => {
+  const { body: { caseIds }, user: { organization_id } } = req;
+  let { query: { type } } = req;
+
+  type = type || 'file'
+
+  if (!caseIds) throw new Error('Case IDs are invalid.')
+  if (!organization_id) throw new Error('Organization ID is not valid.')
+
+  const organization = await organizationsService.fetchById(organization_id);
+  if (organization) {
+    const publishResults = await casesService.publishCases(caseIds, organization.id)
+
+    const points = await casesService.fetchAllPublishedPoints()
+    if (points && points.length > 0) {
```

Comment:
Seems like we're doing a lot in this endpoint. A little concerned running all these operations on a large data set could take an extended period of time and the client might end up timing out. Not anything that we need to handle for right now but curious to see what comes up in testing.

Reply:
On point.  I could see something that Publishes everything, then a cron goes in and generates the Points and files.  It should be pretty fast, but depending on what the testing shows, we may need to refactor a bit.